src/targets/mobile/containers/RatingModal.jsx

```diff
@@ -0,0 +1,236 @@
+    this.props.dontShowAgain()
+  }
+
+  render () {
+    const { screen } = this.state
+
+    return screen === SCREEN_ENJOY ? <EnjoyCozy onReply={this.onUserReply.bind(this)} /> : <FeeedbackForm onClose={this.onCloseFeedback.bind(this)} />
+  }
+}
+
+// sub-components
+class EnjoyCozy extends Component {
+  render ({ onReply }) {
+    const { t } = this.context
+    return (
+      <Modal
+        title={t('mobile.rating.enjoy.title')}
+        withCross={false}
+      >
+        <ModalContent>
+          <div className={styles['button-block']}>
+            <button className='coz-btn coz-btn--secondary' onClick={() => onReply(false)}>
+              { t('mobile.rating.enjoy.no') }
+            </button>
+            <button className='coz-btn coz-btn--regular' onClick={() => onReply(true)}>
+              { t('mobile.rating.enjoy.yes') }
+            </button>
+          </div>
+        </ModalContent>
+      </Modal>
+    )
+  }
+}
+
+class FeeedbackForm extends Component {
+  sending = false
+
+  async onSubmit (e) {
+    e.preventDefault()
+
+    const { t } = this.context
+    const mailData = {
+      mode: 'from',
+      to: [
+        { name: 'Support', email: FEEDBACK_EMAIL }
+      ],
+      subject: t('mobile.rating.email.subject'),
+      parts: [
+        { type: 'text/plain', body: this.textarea.value.toString() }
+      ]
+    }
+
+    try {
+      this.setState({ sending: true })
+      await cozy.client.jobs.create('sendmail', mailData)
+    } catch (e) {
+      // Sending the email failed; this can happen because of insuficient permissions for example. Not a big deal either in this context.
+    }
+
+    this.setState({ sending: false })
+    this.props.onClose(true)
+  }
+
+  registerElement (element) {
+    this.textarea = element
+  }
+
+  render ({ onClose }, { sending }) {
+    const { t } = this.context
+    return (
+      <Modal
+        title={t('mobile.rating.feedback.title')}
+        withCross={false}
+      >
+        <ModalContent className={styles['feedback-form']}>
+          <form className={'coz-form'} onSubmit={this.onSubmit.bind(this)}>
+            <textarea className={styles['feedback-text']} ref={this.registerElement.bind(this)}>
+              { t('mobile.rating.email.placeholder') }
+            </textarea>
+            <div className={styles['button-block']}>
+              <button className='coz-btn coz-btn--secondary' type='button' onClick={() => onClose(false)}>
+                { t('mobile.rating.feedback.no') }
+              </button>
+              <button className='coz-btn coz-btn--regular' type='submit' aria-busy={sending}>
+                { t('mobile.rating.feedback.yes') }
+              </button>
+            </div>
+          </form>
+        </ModalContent>
+      </Modal>)
+  }
+}
+
+// promptRating is not a component because the native UI is used
+const promptRating = ({ title, message, yes, no, later, softwareName, softwareID }) => {
```

Comment:
Do you think it looks a way more better, and if there was no AppRate plugin in it, this function could be easily tested because it is given everything it needs.